sdks/python/src/opik/message_processing/batching/sequence_splitter.py

```diff
@@ -0,0 +1,58 @@
+import json
+from typing import List, Optional, TypeVar, Sequence
+from opik import jsonable_encoder
+
+T = TypeVar("T")
+
+
+def _get_expected_payload_size_MB(item: T) -> float:
+    encoded_for_json = jsonable_encoder.jsonable_encoder(item)
+    json_str = json.dumps(encoded_for_json)
+    return len(json_str.encode("utf-8")) / (1024 * 1024)
+
+
+def split_into_batches(
+    items: Sequence[T],
+    max_payload_size_MB: Optional[float] = None,
+    max_length: Optional[int] = None,
+) -> List[List[T]]:
+    assert (max_payload_size_MB is not None) or (
+        max_length is not None
+    ), "At least one limitation must be set for splitting"
+
+    if max_length is None:
+        max_length = len(items)
+
+    if max_payload_size_MB is None:
+        max_payload_size_MB = float("inf")
+
+    batches: List[List[T]] = []
+    current_batch: List[T] = []
+    current_batch_size_MB: float = 0.0
+
+    for item in items:
+        item_size_MB = (
+            0.0 if max_payload_size_MB is None else _get_expected_payload_size_MB(item)
+        )
+
+        if item_size_MB >= max_payload_size_MB:
+            batches.append([item])
+            continue
```

Comment:
Are we okay to change the order of the dataset items here?

I mean, if the `current_batch` has items and we automatically add the item that exceeds the size as a single batch before here, the order will be different that the actual sent, I guess yes but just double-checking :)

Reply:
I talked to @thiagohora about the order of items in the batch a while ago, but yes, we can double-check in case something has changed.

Reply:
The BE has no restrictions regarding the order of the items. To be fair, by default, the ID actually defines the order of the items when returning them. Anyway, we just persist the item's list, so not a problem.